test/integration/directory_artifact_test.js

```diff
@@ -79,4 +83,86 @@ suite('Directory artifact', () => {
       'Size of uploaded contents of test.html does not match original.'
     );
   });
+
+  // Avoid iptables on local environment
+  if (process.env.WORKER_CI) {
+    test('retry directory upload', async () => {
+      this.timeout(480000);
+
+      let ARTIFACT_COUNT = 3;
+
+      let worker = new TestWorker(DockerWorker);
+      await worker.launch();
+
+      var count = 0;
+
+      let rejectConnection = function() {
+        // Delay iptables rejection to catch
+        // in the middle of the upload
+        setTimeout(function() {
+          iptables.reject({
```

Comment:
I thought I left a comment about this previously related to iptables and teardown of a test... but I think somehow this is failing and then leaving around the ip table rule that breaks after tests after that with "Error: connect ECONNREFUSED "

See chunk 2 of the latest CI run.